cmd/installer/pkg/install/manifest.go

```diff
@@ -227,7 +254,13 @@ func NewManifest(label string, sequence runtime.Sequence, bootPartitionFound boo
 		PartitionType:  LinuxFilesystemData,
 		FileSystemType: FilesystemTypeXFS,
 		Size:           0,
-		Force:          true,
+	}
+
+	if opts.Force {
+		ephemeralTarget.Force = true
+	} else {
+		ephemeralTarget.Skip = true
```

Comment:
handling of upgrades with `--preserve`: don't touch ephemeral, and make last partition before ephemeral use all the space towards the ephemeral